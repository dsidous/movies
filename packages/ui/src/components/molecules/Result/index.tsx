import React, { Fragment } from 'react';
import { compose, branch } from 'recompose';
import { withMovies, withTvs } from '@movies/common';

import MovieCard from '../MovieCard/';

import useStyles from './styles';
import { Show } from '@typesRoots/show';

interface Props {
  shows: Show[];
  resultMedia: string;
  loading: boolean;
}

const Result: React.FC<Props> = ({ shows, resultMedia, loading }) => {
  const classes = useStyles();

  return (
    <ul className={classes.root}>
      {(loading ? Array.from(new Array(6)) : shows).map((movie, index) => (
        <Fragment key={movie?.id || index}>
          {movie ? (
            <MovieCard movie={movie} media={resultMedia} />
          ) : (
            <MovieCard />
          )}
        </Fragment>
      ))}
    </ul>
  );
};

const enhancedComponent = compose(
  branch(
    (props: Props) => props.resultMedia === 'movie',
    withMovies(),
    withTvs()
  )
)(Result);

export default enhancedComponent;
