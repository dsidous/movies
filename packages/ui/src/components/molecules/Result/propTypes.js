import { array, object, string } from 'prop-types';

export const propTypes = {
  config: object.isRequired,
  shows: array.isRequired,
  resultMedia: string.isRequired,
};

export const defaultProps = {};
