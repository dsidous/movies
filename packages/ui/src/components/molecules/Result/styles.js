import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      margin: '0 auto',
      padding: 0,

      [theme.breakpoints.up('sm')]: {
        display: 'grid',
        gridGap: theme.spacing(5),
        gridArea: 'result',
        gridTemplateColumns: 'repeat(2,minmax(200px, 1fr))',
        margin: '0 auto',
      },
    },
  }),
  { name: 'result' },
);
