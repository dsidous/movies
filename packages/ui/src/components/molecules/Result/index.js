/* eslint-disable camelcase */
import React from 'react';
import { compose, branch } from 'recompose';
import { withMovies, withTvs, withConfig } from '@movies/common';

import { propTypes } from './propTypes';
import MovieCard from '../MovieCard';
import Spinner from '../../atoms/Spinner';

import useStyles from './styles';

const Result = ({
  shows,
  config: {
    images: { secure_base_url, poster_sizes },
  },
  resultMedia,
  loading,
  configLoading,
}) => {
  const classes = useStyles();
  const img_base_path = secure_base_url + poster_sizes[3];

  if (loading || configLoading) {
    return <Spinner />;
  }

  return (
    <ul className={classes.root}>
      {shows.map(movie => (
        <MovieCard
          key={movie.id}
          movie={movie}
          img_base_path={img_base_path}
          media={resultMedia}
        />
      ))}
    </ul>
  );
};

Result.propTypes = propTypes;

const enhancedComponent = compose(
  branch(props => props.resultMedia === 'movie', withMovies(), withTvs()),
  withConfig(),
)(Result);

export default enhancedComponent;
