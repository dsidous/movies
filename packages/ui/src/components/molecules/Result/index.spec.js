import React from 'react';
import { shallow } from 'enzyme';
import Result from '.';

const mockProps = {
  shows: [
    {
      id: 1,
    },
  ],
  config: {
    images: {
      secure_base_url: 'path/to/image',
      poster_sizes: ['1', '2', '3', '4'],
    },
  },
  media: 'movie',
};

describe('Molecules/Result', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Result {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
