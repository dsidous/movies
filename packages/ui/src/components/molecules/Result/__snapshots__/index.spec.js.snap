// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Result should render as expected 1`] = `ShallowWrapper {}`;
